src/pages/MenuPage.js

```diff
@@ -155,13 +157,16 @@ export class MenuPage extends React.Component {
 
 export default MenuPage;
 
-/* eslint-disable react/require-default-props, react/forbid-prop-types */
 MenuPage.propTypes = {
-  database: PropTypes.object.isRequired,
+  database: UIDatabaseType.isRequired,
   isInAdminMode: PropTypes.bool,
   logOut: PropTypes.func.isRequired,
   navigateTo: PropTypes.func.isRequired,
-  settings: PropTypes.object.isRequired,
+  settings: PropTypes.shape(SETTINGS_TYPES).isRequired,
```

Comment:
this basically does nothing and the `SETTINGS_TYPES` is a bit confusing. `settings` should be an instance of `MobileAppSettings` which is a singleton?

Reply:
Yeah, it should just be an `instanceOf`. Not sure what I was going for here!

a236a5c46c1873fbe97aabfae62f2a9433903e99.